index.js

```diff
@@ -247,23 +247,28 @@ function refreshMyApps() {
   });
 }
 
-var connectMyDeviceBtn = document.getElementById("connectmydevice");
-
 function getInstalledApps() {
   showLoadingIndicator();
   // Get apps
   Comms.getInstalledApps().then(appIDs => {
     appsInstalled = appIDs;
-    connectMyDeviceBtn.style.display = 'none';
+    handleConnectionChange(true);
     refreshMyApps();
     refreshLibrary();
   }).catch(err => {
     showToast("Getting app list failed, "+err,"error");
   });
 }
 
+var connectMyDeviceBtn = document.getElementById("connectmydevice");
+
+function handleConnectionChange(connected) {
+  connectMyDeviceBtn.style.display = connected ? 'none' : '';
```

Comment:
Would make sense to clean the library on disconnection?

Reply:
I don't know... Maybe if it was an accidental disconnection it makes sense to keep it?

Reply:
Ok, also the implementation may rise the complexity bar of the script